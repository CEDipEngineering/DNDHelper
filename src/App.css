.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.dieDiv {
  display: flex;
  padding: 1rem;
  margin: 1rem auto;
  width: 100%;
  height: fit-content;
  /* background-color: pink; */
  justify-content: center;
}

#diceHeader {
  text-align: center;
}

#Dice {
  /* background-image: url(https://i.ibb.co/HzKKBMg/Clipart-Key-339726.png); */
  background-size: cover;
  width: 75px;
  height: 100px;
  margin-left: 5px;
  margin-right: 5px;


}


/* Images from https://thenounproject.com/ */

.d20{
  background-image: url(https://i.ibb.co/HzKKBMg/Clipart-Key-339726.png);
  background-size: 75%;
  background-repeat: no-repeat;
  margin-top: 15px;
}

.d12{
  background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOEAAADhCAMAAAAJbSJIAAAAflBMVEX///8zMzMsLCxoaGhtbW0tLS0wMDAYGBgpKSkmJiYbGxsiIiIdHR0TExMgICAXFxd6enrk5OTHx8fx8fFhYWH09PTNzc2JiYnb29vV1dV+fn49PT1JSUm+vr6cnJyWlpampqZTU1NaWloKCgo5OTm3t7ekpKSPj49CQkJ0dHTYbg1FAAAMfUlEQVR4nO1d53rCOgwlgTg7jLAhNKzSvv8L3sDtkBw7wZYTeu/n86+lNT5eGpbkwcDCwsLCwsLCwsLCwsLCwsLCwsLCwsLCwsLCwuKFKLZDXaw2r+78EyhYEri6CNJh/moCLVgfI+ZQEMTXV3NoQj5JPRK/Csz/w9P4HgdUfo9pDItXMxFj5sQm+FVg4+Hi1WzqmM592gZE+HvTuFilrjl+FVg6+lPTWAahUX53BO7fmcblPjG4QH/A0s+/MY35aNwFvzsCd/ZqdhXeIqGE0FFo6gPF0u2r+V2ZcAMG4UgDe4G0Cd3lK/mtT0IJ4Z0/9bSSt/PfmsZ8chapaCw5ahsIa+E0ei+aRomKFoYkzfmWiqZxZarTCljvMhE/N71RG/4QTaOzNtJrlX4Ihvrek9GU3vabQDtiad8UL0KC3iedYL4s94LG2c5ArxXwLrEi3Cg8rq66w50vDyMnSULh6MXvRhm0YJGKCT4GO4j96LgtN0oaVz57H+3SJBRI/W+kfapwkzZDt6IZRZfJ+/IJuZjPDiM3aiT3QDDpntg31g1TCGm6YTLejQ4z6easyA1343ZyX5PY32FzVPDF3GmmyfCt4GjeZy5uXpY8vGNfBIvoeYLfnQuzMTvdrg+a02rmLn5FTrWVqC+DkekZS8wLYz89jkK1mYMtOP0QfCP5mzyKtzF+64Ng7qMOB15X5u8DLEAj4vfhSR1BSeEOb0fPz0KyG1hILvbD/WoO2w4+uye4QZLifB/TafE21N9cInjBfcdWylEl5PMz/CTt/uYG6Yzx4ef31QE52lVyOyDSvJMLkOZ3gPue7bsmeE3g1/FnW6U1Ty7nKNajCWcOwoGtJV3f28Tw23yhfFpsyu0x9ONAYXPe5aX7LS95FPBsY3G3BG/Q79SkYyzW19sxi6ozqGU6KymZRfHpxus8EEiHCqkmdiOmSFKMW/XE+xnkSM+gO7nUm9cUuhrWY/hvvgEjW4o5VLSCJ31gdw30bvUhJe0+cxU5uVKOsEUSak5g0IIZkhSZir1WnUGfgKJ3arA46lggn1DanevtAy62rFT9d7DWMkVvfQkpsg/Vb9b7GmW/CXQMZKpWwo42uM9hgYymsfKdSQ4Zqv73DB02UTcODep2hwwT5fFBh1zYiYcYuy5S9SN7CqYhUT4spvjru3BonJDY1RhEGsPBCikbJ/XvbwNWnQKNjQDVhUTdRFgE7QojCRey+gsZRhpGUImU/otGDxqBTRgtgbSGDHX20YfEcDOCHI6fM9ZSKtZA2vg6DJdIYiRmHRqfSFKMtNrYUBkORkgtNuoCx66LVG/0EEMt+yDH3TDp0DBioG3ASo/0LKCnzVNVXKG+xphmK0s6Q+yMjsw5NEIj7ZpgiMc61OxJDYbWhgmG3H4x5AKfYptCe3/PTDDcYBvDjENjaOiMhgz1hRmWW0Pt3gAYk7OzzEQzJnQPDjIntzIKIwzNu8DLRie3ChBDgpGOXeAloUcPLJCkGFNsFlMMC7htWEh1aGC7k+SpvMZmGA7mVFscAvsOzqTDGTLMSJ3C9200iYH9P7RAT2MMB1u4sGgu8BlyXRCXfGmMIT4cfEos+MWkH7YEI0+8IsO+aYJDAwXOkH3pBhlih0akP4lI5VZ2U/N4N8gQ6keUG0UUoEd2UR4AQ6rdg5y3BF15gq78YqJjBDIMaE1NUAqEa4qhk9DMsbffbjGP1hLSvs3NYSVcSW4DyJAUoHblgj8NMqTdvoJzi+SxXvLRrSYZsoSgIt1+zy1KWPq0lidnkmG1vPT1mhVgqC9bF04trsMoQ8fVNzkhQ/1W9vU+URgKAtZDbdeIEYZDQaqcYYZOpmuQgVtypuuSXIlSkUwzdPxSrzXAUFc/Kn1Rh4wzdM56GiqdYXEW9sc8Q737TeR8cLcaZ/JakiPQAUMWqzsDiwtqLIiU05jyWBLm2AFDx1OVZ4KKJ9lOcbF/yOJVu2DoBEobKf8UVTxh6VxFQzrJO2OGIZdGomJK3cT57PeE09XT23GCszxgd8wwZKc9noboWf9+Y0GJp5OGD/iU8fanX4pmGHrDnEt8fM6Umu2yxjBolnw8Y69wBhML8+HvgBtiOB9sOGn0RKTAet5eUIKlo9aTecMZTOcNlD3GGA4KbiDb7icl+fo1uOcW58GUO4jTYtANw8EBK00tptRb9nRJpZA1LXneYPLvR0A3DAdbfKC5DQr01RUfMKGw0AuLjnI9iTOY4sflQkcMB3M8LVJTSlaxJozLjfgjbzyRbEfOYAq+etIRw8EHHk+xKTUdiusRuOntvq6vYvkRJEJFjjOYvk3wzhjmXA6MwJRabMfC/Fcv/Sm4sIqEfxFf6oocZzAxL++YIYqhvCPlO3WIhTNU7TQgXaZz4Syz8Yk7n2ecnPgxa7pjWPtOHGZ4ldRsiy/cBfnyQ1I5BClyOB8IjidkqH+vKWQ4KDmx6P6eEBtJ0cTQF2yx0pdUfyl//iTn0vzS34+MM4TxkjfsVPe+nZ/VASOU8O5YrGBXG1ZYwSfbfStyO/x58nvJBEOqCQxhhgUMPxqJzu+F5Pjw0qFU95EUymTjf/+Fl00gchcGahEYwtsilCRz5GTw5H6HLTtgGjXrQrxvH4V8JnL9AqXQhPphTEt0jQ888bwela3E50bV/VYL5CC2HsMLp0AhHRGlQVFCv6T5VFPOZyJO/A2SZ0ZX7AGoTjD8YwzWOjHHDADFYjgwFoO3ZwRwx8861DbH9qqE0FZboA1BisVoiKfhL/F4sIYDpo5r2FJZEtnbJlNKuZgoKNwPQv/zN79kr7g5bn5TqRMfLve1yZiopnyqrbwCSByo3xbL1PVHe0gekHPMELjwI6R6zSUWbpDpHd/Lj0TSIFqIhclArUFzEQWhj9ZNt9rhsWUm2o6c/9l4iYWGGGFebXQebl5K8uNiVZccLEAjZr5MBg4MwOHLa25ZPekbbEK94HKChgzHeZtJXEcFabgCMTN0rDHPRBZLcUHTyN3loVj9QC/HjEdjvgUMqTSW9QhVqbhEH3WSbwFDfeo5M3BIx2a+EAlhLj6sk5yZikVD3hO0IQlKPgQKgMMeCpz3RIyOk7aLc9dgJoyhcg4weJRL3O8qd60x/xAuYb3UUA7I14UDUTvLP2zMIV3BZWpiY8BtHyCdLJdryWSg22CcBwyTX40UxYM6C85qN5VjJkKOTHhsVMNwd5+em7sBixTLH5xjlhkuTod1JWRVg5hDblVpAa76AG35nVx/NAFcU6EEn0CvrYFlChcpqkTVcU2FproYcGwjqqK4hNsarpXO62I02J1IQlNLxG9lR3PntU0a6tMgAeYTvwa1BRZpD/Vp8CAi/89eroSoQqoi9VBjqKFOFMyGcWn1N5GaC65zcJ0oUvpiA6R+WLSCYsq3I08o3Am91PpqqNcG9VblQmUQBVikUO9876deG+fQADX3oB1MypGHehmwfRcduC7EkNVNzOEuGesv0wU8Sce/allvdRPltS+hsCS8fXcFixGIvB5rX0oNNJjjS8j4nosHqsf6pdIatGijjHW1frTYo59WOCc3mUMzZC7wkfiIUIPkwOq3jrDMBQ4LJmi7F8RCp+da0KJ63g/AX2ueBegcS79/23s9b5kLHFaYDfUeTCmFyt9ng8u9G+C6+j9jilRmvaV0FCnweM30Ulefc4H/kIH1nrXuZaF2++vsRQH0/byNwLllg/3XLG6pzm+Rq3uzh2tUu5CaKvAbJd5XJDpyP+joxh81Z0g+wreJvb1Rwr8z8xU2A+/DNZzfdVf3IcHX6P29M1N/Kyj2Cs75ra5a8a7uwuMjIfp8mKyWEMX80xReCGt4FaHmkqyn9ZcV+3zvSfRml5euLuBHZec3dHU7F8F1fr9vdmFN6gso4kfZ+b0KpG39i6zXd9c4v4kAygd7Pc2ea7Dnt/MGU8GLoQiKzu9lyytZjFaFSwdT8RuWP1B0fm8l8VVfyHa9Exy0PhWftrcA0BQG6AT9vn34i3zblJum5PyeSSLa7vDO+mFkZEjeA35AyfktzziuJG3vj6wiFEwag8kUXnOWTmDcmLfXDyTvcjtKT3NLWgiinsylZnT2tjobt+fP9gRJSiGRn3IgdaeQpBQSEAblq0lhLFap8qObDVCpRdAb6qGv2mC+Uj2J/jC7kN7v/EHsUEs0docWRe4pvExFew6LrTgF8Wl4qSyl+89gfSRsx8a0/L+DwkkCBXXmF0HCenMXElGshjrY/lf4WVhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFj8T/EPxpbEHLdGnmIAAAAASUVORK5CYII=);
  background-size: cover;
}


.d10{
  background-image: url(https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTeOsJWJUC1YMJ7moWTAqgnrHgFWaiu5DKF4w&usqp=CAU);
  background-size: 85%;
  background-repeat: no-repeat;
  background-position: center;
}

.d4{
  background-image: url(https://i.ibb.co/cJ6GG9n/d4.png);
  background-size: 85%;
  background-repeat: no-repeat;
  background-position: center;
}

