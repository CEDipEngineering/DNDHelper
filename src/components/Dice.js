import React, { Component, Fragment } from 'react'
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';
import AddIcon from '@material-ui/icons/Add';
import RemoveIcon from '@material-ui/icons/Remove';

class Dice extends React.Component {
    constructor(props) {
        super()
        this.state = {
            loading: true,
            // setOpen: false,
            count: 1,
        }
        this.number = ""
        this.roll = this.roll.bind(this);
        this.upCount = this.upCount.bind(this);
        this.dropCount = this.dropCount.bind(this);
        this.handleClick = this.handleClick.bind(this);
    }

    roll() {
        return Math.floor(Math.random() * this.props.number) + 1
    }

    handleClick(){
        let curr = 0;
        for (let i = 0; i<this.state.count; i++){
            curr += this.roll();
        }
        this.number = curr;
        this.setState(
            {setOpen: true}
        )
    };

    componentDidMount() {
        this.setState({ loading: false })
    }

    upCount(event){
        this.setState({
            count: this.state.count + 1
        });
    }

    dropCount(event){
        this.setState({
            count: this.state.count - 1
        });
    }

    render() {
        return (
            <React.Fragment>
                <div className={"dieSelector"}>
                    <h5>{this.state.count}</h5>
                    <IconButton onClick={this.upCount}><AddIcon/></IconButton>
                    <IconButton onClick={this.dropCount}><RemoveIcon/></IconButton>
                </div>
                <Tooltip title={"Dice " + this.props.number}>
                    <IconButton id={"Dice"} onClick={this.handleClick}></IconButton>
                </Tooltip>
                <p>Result: {this.number}</p>
            </React.Fragment>
        )
    }

}
//  Placeholder image of Dice
// https://i.ibb.co/HzKKBMg/Clipart-Key-339726.png

export default (Dice)