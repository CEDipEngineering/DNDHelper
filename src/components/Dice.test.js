import React from "react";
import { shallow } from "enzyme";
import Dice from "./Dice";
import Enzyme from "enzyme";
import { Snackbar, IconButton } from "./Dice";
import Login from "../main";
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from "../App";
import Main from '../main'

configure({ adapter: new Adapter() });


describe('Dice', () => {
it("component renders correctly", () => {
  const wrapper = shallow(
    <Dice />
  );
  expect(wrapper).toMatchSnapshot();
});


it("random function works correctly for dice 20", () => {
  
  const wrapper = shallow(
      <Dice {...{"number": 20}}/>
    );

  let value;
  for (let i=0; i<100;i++){
    value = wrapper.instance().roll()
    expect(value).toBeGreaterThanOrEqual(1);
    expect(value).toBeLessThan(wrapper.instance().props.number+1);
  }
  
  });
  
  
it("random function works correctly for dice 12", () => {
  
  const wrapper = shallow(
      <Dice {...{"number": 12}}/>
    );

  let value;
  for (let i=0; i<100;i++){
    value = wrapper.instance().roll()
    expect(value).toBeGreaterThanOrEqual(1);
    expect(value).toBeLessThan(wrapper.instance().props.number+1);
  }
  
  });
  

 });
