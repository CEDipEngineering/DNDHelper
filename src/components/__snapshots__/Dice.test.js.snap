// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dice component renders correctly 1`] = `ShallowWrapper {}`;

exports[`component renders correctly 1`] = `ShallowWrapper {}`;
